### PR TITLE
Try update version and deps for v0.21

### PR DESCRIPTION
**Pull request by jnothman**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jnothman** [comment]
@conda-forge-admin, please rerender

**Comment by jnothman** [comment]
@conda-forge-admin, please rerender

**Comment by jnothman** [comment]
I'm lost here. I thought that updating `min_py_ver` in `conda-forge.yml`, and rerendering would make `.ci_support/*python2.7*` disappear. But rerendering makes no change.

**Comment by jnothman** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -9,10 +9,10 @@ package:
 source:
   fn: {{ version }}.tar.gz
   url: https://github.com/scikit-learn/scikit-learn/archive/{{ version }}.tar.gz
-  sha256: cadeefa9a824671bdcad790a075efbaa0d5a00458c43c9b4ce7110b129df7b10
+  sha256: 15ff8f24909c68340532147ca6ff68e05508c836ceb5307c23d80a825a3e2a45
 
 build:
-  number: 1
+  number: 0
```
Comment (rth) [inline recipe/meta.yaml]: You need to add here,
```
 skip: true  # [py<35]
```
then re-rerender, as the minimal python versions is not parsed from `setup.py` I think.

**Comment by jnothman** [comment]
Failures on Python 3 are like:
```
ImportError: Your installation of Numerical Python (NumPy) 1.9.3 is out-of-date.
scikit-learn requires NumPy >= 1.11.0.
```

Where is the right place to indicate minimum numpy dependency?

**Comment by jnothman** [comment]
Closing in favour of #93

**Pull request closed without merging**